xcube_cci/cciodp.py

```diff
@@ -1092,6 +1107,18 @@ async def _extract_metadata_from_odd_url(self, session: aiohttp.ClientSession,
         xml_text = await resp.read()
         return _extract_metadata_from_odd(etree.XML(xml_text))
 
+    def _maybe_adjust_data_type(self, dtype: str) -> str:
+        adjusted_dtype = _DTYPES_TO_DTYPES_WITH_MORE_BYTES.get(dtype, dtype)
```

Comment:
Using  `dtype` as default may cause the problem I was referring to: accidently masking out valid data. I suggest letting the call fail instead same as for the 64-bit ints (you could then also remove entry `'(u)int64'` from `_DTYPES_TO_DTYPES_WITH_MORE_BYTES`):

```suggestion
        adjusted_dtype = _DTYPES_TO_DTYPES_WITH_MORE_BYTES.get(dtype)
```

Reply:
Well, we need to consider floating point data types, too, which will not change. But I see how we can improve this bit.